moped-editor/src/views/projects/projectView/MilestoneTemplateModal.js

```diff
@@ -0,0 +1,177 @@
+import React, { useState, useEffect } from "react";
+import {
+  Button,
+  Box,
+  Checkbox,
+  Dialog,
+  DialogTitle,
+  DialogContent,
+  IconButton,
+  List,
+  ListItemText,
+  ListItemIcon,
+  ListItem,
+  TextField,
+} from "@material-ui/core";
+import { Autocomplete } from "@material-ui/lab";
+import { makeStyles } from "@material-ui/core";
+import CloseIcon from "@material-ui/icons/Close";
+import AddCircle from "@material-ui/icons/AddCircle";
+import { returnSignalPHBMilestoneTemplate } from "../../../utils/timelineTemplates";
+
+import { ADD_PROJECT_MILESTONE } from "../../../queries/project";
+import { useMutation } from "@apollo/client";
+
+const useStyles = makeStyles((theme) => ({
+  dialogTitle: {
+    color: theme.palette.primary.main,
+    fontFamily: theme.typography.fontFamily,
+    display: "flex",
+    justifyContent: "space-between",
+    alignItems: "center",
+  },
+}));
+
+const templateChoices = ["AMD", "PDD"];
+
+const MilestoneTemplateModal = ({
+  isDialogOpen,
+  handleDialogClose,
+  projectId,
+  milestoneNameLookup,
+  selectedMilestones,
+  refetch,
+}) => {
+  const classes = useStyles();
+
+  const [template, setTemplate] = useState(null);
+  const [milestonesList, setMilestonesList] = useState([]);
+  const [milestonesToAdd, setMilestonesToAdd] = useState([]);
+
+  const [addProjectMilestone] = useMutation(ADD_PROJECT_MILESTONE);
+
+  useEffect(() => {
+    let options = [];
+    const selectedMilestonesIds = selectedMilestones.map(
+      (milestone) => milestone.milestone_id
+    );
+    setMilestonesList([]);
+    if (template === "AMD") {
+      options = returnSignalPHBMilestoneTemplate(projectId);
+    }
+    // if (template === "PDD") {
+    // options = return other template
+    // }
+    const filteredOptions = options.filter(
+      (option) => !selectedMilestonesIds.includes(option.milestone_id)
+    );
+    setMilestonesList(filteredOptions);
+    setMilestonesToAdd(filteredOptions);
+  }, [template, selectedMilestones, projectId]);
```

Comment:
this is tricky! i think you can simplify the hooks/state by separating out the milestoneList into a memo.

```javascript
// outside the main component
const useMilestoneOptions = (template) =>
  useMemo(() => {
    if (!template) {
      return [];
    } else if (template === "AMD") {
      return returnSignalPHBMilestoneTemplate(projectId);
    } else if (template === "PDD") {
      // etc
      return [];
    }
  }, [template]);
```

then you can trim your effect to just (re)setting the checked milestones
```javascript
// inside component
 const milestonesList = useMilestoneOptions(template)

  useEffect(() => {
    setMilestonesToAdd([...milestonesList]);
  }, [milestonesList]);
```

beyond that, your `handleToggle` should take care of the rest. i might be oversimplifying it, but breaking this into a smaller hooks will probably weed out tricky bugs.